lib/auth/mfa/mfav1/service.go

```diff
@@ -16,18 +16,439 @@
 
 package mfav1
 
-import mfav1 "github.com/gravitational/teleport/api/gen/proto/go/teleport/mfa/v1"
+import (
+	"cmp"
+	"context"
+	"log/slog"
+
+	"github.com/gravitational/trace"
+
+	"github.com/gravitational/teleport"
+	"github.com/gravitational/teleport/api/client/proto"
+	mfav1 "github.com/gravitational/teleport/api/gen/proto/go/teleport/mfa/v1"
+	"github.com/gravitational/teleport/api/types"
+	apievents "github.com/gravitational/teleport/api/types/events"
+	wanlib "github.com/gravitational/teleport/lib/auth/webauthn"
+	wantypes "github.com/gravitational/teleport/lib/auth/webauthntypes"
+	"github.com/gravitational/teleport/lib/authz"
+	"github.com/gravitational/teleport/lib/events"
+)
+
+// AuthServer defines the subset of Auth server methods used by the MFA service.
+type AuthServer interface {
+	BeginSSOMFAChallenge(
+		ctx context.Context,
+		user string,
+		sso *types.SSOMFADevice,
+		ssoClientRedirectURL,
+		proxyAddress string,
+		_ *mfav1.ChallengeExtensions,
+		sip *mfav1.SessionIdentifyingPayload,
+	) (*proto.SSOChallenge, error)
+
+	VerifySSOMFASession(
+		ctx context.Context,
+		username,
+		sessionID,
+		token string,
+		_ *mfav1.ChallengeExtensions,
+	) (*authz.MFAAuthData, error)
+}
+
+// Cache defines the subset of cache methods used by the MFA service.
+type Cache interface {
+	GetAuthPreference(ctx context.Context) (types.AuthPreference, error)
+	GetClusterName(ctx context.Context) (types.ClusterName, error)
+}
+
+// Emitter defines the subset of event emitter methods used by the MFA service.
```

Comment:
Suggestion: Same here and for Identity: call out the main implementer, so it's easy for the uninitiated to track it.